Integrations/integration-CiscoUmbrellaInvestigate.yml

```diff
@@ -0,0 +1,226 @@
+    400 Bad Request—Likely missing a required parameter or malformed JSON. Please check the syntax on your query.
+    403 Unauthorized—Request had Authorization header but token was missing or invalid. Please ensure your API token is valid.
+    404 Not Found—The requested item doesn't exist, check the syntax of your query or ensure the IP and/or domain are valid.
+    500, 502, 503, 504 Server errors—something went wrong on our end.
+     */
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        return res;
+    };
+
+    switch (command) {
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            return 'ok';
+        case 'domain-category':
+            /*
+            Returned Value For Output If Success 200
+    Field    Type    Description
+    status
+    integer
+    The status will be "-1" if the domain is believed to be malicious, "1" if the domain is believed to be benign, "0" if it hasn't been classified yet.
+    security_categories
+    array of strings
+    The Umbrella security category, or categories, that match this domain or that this domain is associated with. If none match, the return will be blank.
+    content_categories
+    array of strings
+    The Umbrella content category or categories that match this domain. If none match, the return will be blank.
+    */
+            //log(JSON.stringify(params));
+            var url = 'https://investigate.api.opendns.com/domains/categorization/' + encodeURIComponent(args.domain) + '?showLabels';
+            var resp = sendRequest(url);
+            return resp;
+            //return {ContentsFormat: 'markdown', Contents: '# Simple test with arg - ' + args.domain, Type: 1};
+      /* DEPRECATED
+
+        case 'domain-score':
+            var url = 'https://investigate.api.opendns.com/domains/score/' + encodeURIComponent(args.domain);
+            var resp = sendRequest(url);
+            return resp;
+    */
+        case 'related-domains':
+            //log(JSON.stringify(params));
+            var url = 'https://investigate.api.opendns.com/recommendations/name/' + encodeURIComponent(args.domain) + '.json';
+            var resp = sendRequest(url);
+            return resp;
+            //return {ContentsFormat: 'markdown', Contents: '# Simple test with arg - ' + args.domain, Type: 1};
+        case 'domain-search':
+            var url = 'https://investigate.api.umbrella.com/search/' + encodeURIComponent(args.domainregex) + '?';
+
+            if(args['start'] && args.start !== '') {
+                url += 'start=' + encodeURIComponent(args.start) + '&';
+            } else {
+                url += 'start=-32days&';
+            }
+            url += 'includecategory=true';
+            var resp = sendRequest(url);
+            return resp;
+        case 'domain-links':
```

Comment:
scripts/script-commonIntegration.yml
it takes a string with %val% mark and an object, if the object has a val key it assign it's value in the string and delets it from the object.
you can use it when you have a http://blabla/get/%id% url and an id key in the args.